eldritch/mythos.py

```diff
@@ -456,13 +473,168 @@ def get_trigger(self, event, state):
+    return seq
+
+
+class Mythos38(Environment):
+  def __init__(self):
+    super().__init__(
+        "Mythos38", "WitchHouse", "Cave", {"slash", "triangle", "star"}, {"hex"}, "mystic"
+    )
+
+  def get_modifier(self, thing, attribute, state):
+    if isinstance(thing, events.CloseGate) and attribute == "seal_clues":
+      return -2
+    return 0
+
+
+class Mythos39(Environment):
+  def __init__(self):
+    super().__init__("Mythos39", "Unnamable", "Woods", {"plus"}, {"moon"}, "mystic")
+
+  def get_override(self, thing, attribute):
+    if attribute == "can_seal":
+      return False
+    return None
+
+
+class Mythos40(Headline):
+  def __init__(self):
+    super().__init__("Mythos40", "Woods", "Society", {"circle"}, {"square", "diamond"})
+
+  def create_event(self, state) -> events.Sequence:
+    seq = super().create_event(state)
+    seq.events.append(events.ReturnToCup(from_places=["Uptown", "Hospital", "Shoppe", "Woods"]))
+    return seq
+
+
+class Mythos41(Headline):
+  def __init__(self):
+    super().__init__("Mythos41", "Lodge", "Graveyard", {"circle"}, {"square", "diamond"})
+
+  def create_event(self, state) -> events.Sequence:
+    seq = super().create_event(state)
+    seq.events.append(events.ReturnToCup(from_places=["Northside", "Shop", "Newspaper", "Train"]))
+    return seq
+
+
+class Mythos42(Environment):
+  def __init__(self):
+    super().__init__("Mythos42", "WitchHouse", "Cave",
+                     {"square", "diamond"}, {"circle"}, env_type="mystic")
+
+  def get_modifier(self, thing, attribute, state):
+    if isinstance(thing, items.Spell) and attribute == "sanity_cost":
+      return -float("inf")
+    return 0
+
+
+class Mythos43(Headline):
+  def __init__(self):
+    super().__init__("Mythos43", "WitchHouse", "Cave", {"square", "diamond"}, {"circle"})
+
+  def create_event(self, state) -> events.Sequence:
+    seq = super().create_event(state)
+    seq.events.append(events.ReturnToCup(from_places=["Rivertown", "Graveyard", "Cave", "Store"]))
+    return seq
+
+
+class Mythos44(Environment):
+  def __init__(self):
+    super().__init__("Mythos44", "Roadhouse", "Square", {"plus"}, {"moon"}, env_type="weather")
+
+  def create_event(self, state):
+    seq = super().create_event(state)
+    seq.events.extend([
+        events.ReturnMonsterFromBoard(None, monster)
+        for monster in state.monsters
+        if isinstance(monster.place, places.CityPlace) and monster.name == "Flame Matrix"
+    ])
+
+  def get_modifier(self, thing, attribute, state):
+    if isinstance(getattr(thing, "place", None), places.CityPlace):
+      if attribute == "speed":
+        return -1
+      if attribute == "sneak":
+        return 1
+    return 0
+
+  def get_interrupt(self, event, state):
+    if isinstance(event, events.CityMovement):
+      return events.ChangeMovementPoints(event.character, -1)
+    return None
+
+  def get_trigger(self, event, state):
+    # TODO: Figure out how to override the FlameMatrix deployment
```

Comment:
I think there probably needs to be a global modifier for invalid monsters (that cannot be drawn from the cup), and the DrawMonstersFromCup event just needs to check it.

Reply:
It's not obvious to me that will do everything we need it to. In the FAQs, 

> Q: If investigators fail [Mythos 65 Rumor] while the [Mythos 44]  environment is in play, would that send [Flame Matrices] that were on the card to the cup?
> A: Yes.

Which implies that a Flame Matrix could also be placed onto Mythos65. 

Also, I think it's not obvious whether drawing a Flame Matrix while in an Other World would still mean you fight it.  Is it raining _so hard_ that it also affects things on Pluto?

Reply:
I like this use of `can_*`.